src/target/riscv/riscv.c

```diff
@@ -2712,6 +2712,13 @@ static int riscv_get_gdb_reg_list(struct target *target,
 
 static int riscv_arch_state(struct target *target)
 {
+	assert(target->state == TARGET_HALTED);
+	LOG_USER("%s halted due to %s.%s",
+			target_name(target),
+			debug_reason_name(target),
+			(target->semihosting && target->semihosting->is_active) ?
+			" Semihosting is active." :
+			"");
```

Comment:
Minor detail for readability of the LOG_USER call. (Optional - if you like.)

```suggestion
	const bool semihosting_active = (target->semihosting && target->semihosting->is_active);
	LOG_USER("%s halted due to %s.%s",
			target_name(target),
			debug_reason_name(target),
			semihosting_active ? " Semihosting is active." : "");
```

Reply:
Addressed.